lib/src/main/java/com/springboot/webpush/common/api/Notification.java

```diff
@@ -4,40 +4,61 @@
 import java.util.List;
 import java.util.Map;
 
+import org.springframework.util.Assert;
 
-public class Notification {
-    //private static final String OPEN_PWA_ACTION = "openPWA";
+import com.springboot.webpush.common.api.types.OnActionClickOperation;
 
-    private final String               title;
-    private final String               body;
-    private final Map<String, Object>  data;
-    private final List<Integer>        vibrate;
-    private final List<Action>         actions;
-    private final String               icon;
-    private final String               image;
-    private final boolean              requireInteraction;
+/**
+ * The {@link Notification} class is the primary message structure, as expected by the browser creators web push
+ * services.
+ */
+public class Notification {
+    private final String              title;
```

Comment:
A lot of these are duplicated in Action.java. Maybe need to de-duplicate? Or maybe it's fine as is, since this builds up the action?

Reply:
We chatted about this -- Action was kind of the wrong class -- renamed it and depreciated it - just to keep it around - but it is not currently used.